BITCODE.cpp

```diff
@@ -0,0 +1,79 @@
+#include <algorithm>
+#include <bitset>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+
+#include <llvm/IR/DebugInfoMetadata.h>
+#include <llvm/IR/Function.h>
+#include <llvm/IR/Instructions.h>
+#include <llvm/IR/IntrinsicInst.h>
+#include <llvm/IR/Metadata.h>
+#include <llvm/IR/Module.h>
+#include <llvm/IR/Type.h>
+#include <llvm/Support/raw_ostream.h>
+
+#include <remill/Arch/Arch.h>
+#include <remill/BC/Util.h>
+#include <remill/OS/OS.h>
+#include <remill/Arch/Name.h>
+
+#include <anvill/Decl.h>
+
+DECLARE_string(arch);
+DECLARE_string(os);
+DEFINE_string(bc_file, "",
+              "Path to BITcode file containing data to be specified");
+
+int main(int argc, char *argv[]) {
+  google::ParseCommandLineFlags(&argc, &argv, true);
+  google::InitGoogleLogging(argv[0]);
+
+  if (FLAGS_bc_file.empty()) {
+    LOG(ERROR) << "Please specify a path to a BITcode input file in --bc_file";
+    return EXIT_FAILURE;
+  }
+
+  // Overwrite the inherited architecture and os flags if they are not
+  // already empty.
+  if (!FLAGS_arch.empty() || !FLAGS_os.empty()) {
+    FLAGS_arch = "";
+    FLAGS_os = "";
+  }
+
+  auto context = new llvm::LLVMContext;
+  auto module = remill::LoadModuleFromFile(context, FLAGS_bc_file);
+  auto arch = remill::Arch::GetModuleArch(*module);
```

Comment:
Can this fail and return `nullptr`? If so, maybe check for it, log a message, and `return EXIT_FAILURE;`

Reply:
Fixed c9cdb7b